### PR TITLE
Create data model

### PR DESCRIPTION
**Pull request by hammaazarok**
## In this milestone I have achieved the following :
 - Create new rails app
 - Initialize the project with Git
-  Use Rails Active Record Migrations
-  Create and run the necessary migration files for User, Comment, Like, and Post
-  Add relations between tables in the ERD diagram to the Database

### CODE REVIEW

File: db/migrate/20221019091238_add_user_ref_to_post.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddUserRefToPost < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :posts, :user, null: false, foreign_key: true
```
Comment (TedLivist) [inline db/migrate/20221019091238_add_user_ref_to_post.rb]: - [x] The posts table should have an `author_id`, not `user_id`, and so this reference should be `author`, not `user`. Kindly run the necessary migration to fix this. You can look at this [StackOverflow answer](https://stackoverflow.com/a/42056089) for ideas on how to specify references for different column names

File: db/migrate/20221019091306_add_user_ref_to_comment.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddUserRefToComment < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :comments, :user, null: false, foreign_key: true
```
Comment (TedLivist) [inline db/migrate/20221019091306_add_user_ref_to_comment.rb]: - [x] Just as suggested above this should point to the `author`, not `user`. Kindly fix that

File: db/migrate/20221019091315_add_user_ref_to_like.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddUserRefToLike < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :likes, :user, null: false, foreign_key: true
```
Comment (TedLivist) [inline db/migrate/20221019091315_add_user_ref_to_like.rb]: - [x] Just as suggested above this should point to the `author`, not `user`. Kindly fix that.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @hammaazarok,

Good job so far!
✔️ Use of Postgres for db
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- [x] **Kindly format your README using the Microverse template**

- [x] _Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20221019091238_add_user_ref_to_post.rb (hunk 1 above)
Comment (hammaazarok) [reply db/migrate/20221019091238_add_user_ref_to_post.rb]: thanks for the review and the valuable resources you provided

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @hammaazarok , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- [ ] All linters passed

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**